src/vulkan-renderer/wrapper/framebuffer.cpp

```diff
@@ -0,0 +1,65 @@
+#include "inexor/vulkan-renderer/wrapper/framebuffer.hpp"
+
+#include <spdlog/spdlog.h>
+
+#include <cassert>
+
+namespace inexor::vulkan_renderer::wrapper {
+
+Framebuffer::Framebuffer(Framebuffer &&other) noexcept : device(other.device), name(std::move(other.name)), frames(std::move(other.frames)) {}
+
+Framebuffer::Framebuffer(const VkDevice device, const VkRenderPass renderpass, const std::vector<VkImageView> &attachments,
+                         const std::vector<VkImageView> &swapchain_attachments, const std::uint32_t width, const std::uint32_t height,
+                         const std::uint32_t swapchain_image_count, const bool multisampling_enabled, const std::string &name)
+    : device(device), name(name) {
+    assert(device);
+    assert(renderpass);
+    assert(!name.empty());
+    assert(!attachments.empty());
+    assert(width > 0);
+    assert(height > 0);
+    assert(swapchain_image_count > 0);
+
+    spdlog::debug("Creating frame buffers.");
+    spdlog::debug("Number of images in swapchain: {}.", swapchain_image_count);
+
+    std::vector<VkImageView> framebuffer_attachments = attachments;
+
+    VkFramebufferCreateInfo frame_buffer_ci = {};
```

Comment:
Better use `VkFramebufferCreateInfo frame_buffer_ci{};`

Reply:
We will do this in another pull request when we introduce initializer functions.